compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/model/symbols/BInvokableSymbol.java

```diff
@@ -40,6 +40,8 @@
 public class BInvokableSymbol extends BVarSymbol implements InvokableSymbol {
 
     public List<BVarSymbol> params;
+    public List<BVarSymbol> includedRecordParams;
+    public BVarSymbol incRecordParamAllowAdditionalFields;
```

Comment:
If we are not keeping these two fields we need to check for an open included record which allows extra fields in typechecker and desugar phase also. We can avoid the effect of repeating the same task through this.

Reply:
In desugar it's just the additional checks for two flags, right? One to see if the parameter has the `INCLUDED` flag and the other to see if the record is open? IMO, that is OK rather than adding these fields just for the desugar phase.

Reply:
Removed `includedRecordParams` and `incRecordParamAllowAdditionalFields` fields.